src/vivarium/framework/population/population_view.py

```diff
@@ -208,46 +205,143 @@ def update(self, population_update: Union[pd.DataFrame, pd.Series]) -> None:
+        to rule out several categories of error in client simulation code. The state table
+        is modified at three different times. In the first, the initial population table
+        is being created and new columns are being added to the state table with their
+        initial values. In the second, the population manager has added new rows with
+        appropriate null values to the state table in response to population creation
+        dictated by client code, and population updates are being provided to fill in
+        initial values for those new rows. In the final case, state table values for
+        existing simulants are being overridden as part of a time step.
+
+        All of these modification scenarios require that certain preconditions are met.
+        For all scenarios, we require
+
+            1. The update is a DataFrame or a Series.
+            2. If it is a series, it is nameless and this view manages a single column
+               or it is named and it's name matches a column in this PopulationView.
+            3. The update matches at least one column in this PopulationView.
+            4. The update columns are a subset of the columns managed by this
+               PopulationView.
+            5. The update index is a subset of the existing state table index.
+               PopulationViews don't make rows, they just fill them in.
+
+        For initial population creation additional preconditions are documented in
+        :meth:`PopulationView._ensure_coherent_initialization`. Outside population
+        initialization, we require that all columns in the update to be present in
+        the existing state table. When new simulants are added in the middle of the
+        simulation, we require that only one component provide updates to a column.
+
+        Parameters
+        ----------
+        population_update
+            The update to the simulation state table.
+        state_table
+            The existing simulation state table.
+        view_columns
+            The columns managed by this PopulationView.
+        creating_initial_population
+            Whether the initial population is being created.
+        adding_simulants
+            Whether new simulants are currently being initialized.
+
+        Returns
+        -------
+        pandas.DataFrame
+            The input data formatted as a DataFrame.
+
+        Raises
+        ------
+        TypeError
+            If the population update is not a :class:`pandas.Series` or a
+            :class:`pandas.DataFrame`.
+        PopulationError
+            If the update violates any preconditions relevant to the context in which
+            the update is provided (initial population creation, population creation on
+            time steps, or population state changes on time steps).
+
+        """
+        assert not (creating_initial_population and not adding_simulants)
```

Comment:
Nit: legibility for the Boolean expression.

`not (creating_initial_population and not adding_simulants)`

vs

`not creating_initial_population or adding_simulants`